fs/expose/sftp.py

```diff
@@ -236,12 +236,14 @@ def close(self):
 
     @report_sftp_errors
     def read(self, offset, length):
-        self._file.seek(offset)
+        if hasattr(self._file, 'seek'):
+            self._file.seek(offset)
         return self._file.read(length)
```

Comment:
What happens if they try to open up a big file with this change? Would it all be read into memory to be served up to the user?